src/main/groovy/edu/oregonstate/mist/osuevents/db/LocalistDAO.groovy

```diff
@@ -0,0 +1,455 @@
+    URI baseURI
+
+    static String localistApiVersion = "2.2"
+
+    static String filtersEndpoint = "/events/filters"
+    static String placesEndpoint = "/places"
+    static String departmentsEndpoint = "/departments"
+    String communitiesEndpoint // set in constructor as config param, depends on organizationID
+
+    ObjectMapper objectMapper = new ObjectMapper()
+
+    private static Logger logger = LoggerFactory.getLogger(this)
+
+    LocalistDAO(HttpClient httpClient, String baseURL, String organizationID) {
+        this.httpClient = httpClient
+        this.baseURI = UriBuilder.fromUri(baseURL.toURI())
+                .path("/api/${localistApiVersion}").build()
+        this.communitiesEndpoint = "/organizations/${organizationID}/communities"
+    }
+
+    /**
+     * Get event topics from filters.
+     * @return
+     */
+    List<EventTopic> getEventTopics() {
+        Filters filters = getFilters()
+
+        filters.eventTopics.collect { EventTopic.fromFilter(it) }
+    }
+
+    /**
+     * Get single event topic by ID.
+     * @param id
+     * @return
+     */
+    EventTopic getEventTopicByID(String id) {
+        Filter filter = getFilters().eventTopics.find { it.filterID == id }
+
+        if (filter) {
+            EventTopic.fromFilter(filter)
+        } else {
+            null
+        }
+    }
+
+    /**
+     * Get event types from filters.
+     * @return
+     */
+    List<EventType> getEventTypes() {
+        Filters filters = getFilters()
+
+        filters.eventTypes.collect { EventType.fromFilter(it) }
+    }
+
+    /**
+     * Get single event type by ID.
+     * @param id
+     * @return
+     */
+    EventType getEventTypeByID(String id) {
+        Filter filter = getFilters().eventTypes.find { it.filterID == id }
+
+        if (filter) {
+            EventType.fromFilter(filter)
+        } else {
+            null
+        }
+    }
+
+    /**
+     * Get audiences from filters.
+     * @return
+     */
+    List<Audience> getAudiences() {
+        Filters filters = getFilters()
+
+        filters.audiences.collect { Audience.fromFilter(it) }
+    }
+
+    /**
+     * Get single audience by ID.
+     * @param id
+     * @return
+     */
+    Audience getAudienceByID(String id) {
+        Filter filter = getFilters().audiences.find { it.filterID == id }
+
+        if (filter) {
+            Audience.fromFilter(filter)
+        } else {
+            null
+        }
+    }
+
+    /**
+     * Get counties from filters.
+     * @return
+     */
+    List<County> getCounties() {
+        Filters filters = getFilters()
+
+        filters.counties.collect { County.fromFilter(it) }
+    }
+
+    /**
+     * Get single county by ID.
+     * @param id
+     * @return
+     */
+    County getCountyByID(String id) {
+        Filter filter = getFilters().counties.find { it.filterID == id }
+
+        if (filter) {
+            County.fromFilter(filter)
+        } else {
+            null
+        }
```

Comment:
Same here?
Actually `getCountyByID`, `getAudienceByID`, `getEventTypeByID` and `getEventTypes` are almost the same, could we combine them into one?

Reply:
I wanted this class to create some separation between counties, audiences, eventtypes, and eventtopics to limit the amount of code in each resource class.